contrib/database/sql/option.go

```diff
@@ -292,3 +293,10 @@ func WithDBStats() Option {
 		cfg.dbStats = true
 	}
 }
+
+func WithDBStatsTags(tags []string) Option {
```

Comment:
```suggestion
// WithDBStatsTags enables DBStats metrics collection and sets the given tags for those metrics.
func WithDBStatsTags(tags map[string]string) Option {
```

Could we make this a `map[string]string`? Since this is how we model tags generally everywhere else.
We can keep `cfg.statsTags` as a `[]string` and do the transformation here.

Reply:
https://github.com/DataDog/dd-trace-go/blob/137578f423ce7c5936e677e10ba92c9f56639125/contrib/database/sql/metrics.go#L60

https://github.com/DataDog/dd-trace-go/blob/c345adb632da7983b294c3995f9e93573d23bcad/internal/statsd.go#L16


That does make more sense, but since your public interfaces accept `[]string` with an only an expected convention of using `:` as the delimiter between keys/values, I think it's probably better to address those public places first and/or update documentation.

I'm closing this as I no longer have the time to spend on it.